addon/components/ember-list.js

```diff
@@ -117,18 +151,20 @@ export default Ember.Component.extend({
 
     for (i=0; i<this.cells.length; i++) {
       cell = this.cells[i];
-      if (!cellMap[cell.index]) {
+      if (!cellMap[cell.key]) {
         if (newItems.length) {
           itemIndex = newItems.pop();
+          itemKey = decodeEachKey(items[itemIndex], '@identity');
           pos = this.cellLayout.positionAt(itemIndex, this.width, this.height);
           width = this.cellLayout.widthAt(itemIndex, this.width, this.height);
           height = this.cellLayout.heightAt(itemIndex, this.width, this.height);
           style = formatStyle(pos, width, height);
           Ember.set(cell, 'style', style);
+          Ember.set(cell, 'key', itemKey);
```

Comment:
@raytiley when an item goes out of view, and we set a different item on that cell, the #each helper is going to teardown the subtree of that cell and rerender it. I'm going to merge this for now, what I had in mind is we'd map item key to cell key, so if you mapped a different item key to the same cell key, it meant you wanted to reuse an unused cell, if you make the cell key match the item key you didn't want reuse.